retis/src/core/events/bpf.rs

```diff
@@ -333,7 +342,8 @@ pub(crate) fn parse_raw_event<'a>(
         cursor += mem::size_of_val(&raw_section.header);
 
         // Compute where the current section ends.
-        let raw_section_end = cursor + raw_section.header.size as usize;
+        let raw_section_end =
+            cursor + raw_section.header.size as usize - raw_section.header.pad_size as usize;
```

Comment:
Removing the pad size if fine when setting up the raw section Rust representation, but the below checks against the real size of the event (pad included):

```rust
        } else if raw_section_end > raw_event_size {                            
            error!("Section goes past the buffer: {raw_section_end} > {raw_event_size}");
            break;                                                              
        }
```

Reply:
good catch. I overlooked this for some reason. 
It's fairly easy to fix this (also most probably this could happen in synthetic tests only), but regardless, we're very close to the event limit and the padding strategy (admittedly conservative on the correctness side) may lead to overflow. We should consider increasing the event size, reducing the modules' requirement (e.g. reducing the nft chain buffer size), optimizing the alignment strategy, or completely changing the raw events (the best option long term, but I don't consider this a priority now).

Reply:
Either way, this got split because of the overflow risk, so it's no longer part of this series.

Reply:
Right, let's tackle the alignment in a separate PR. I believe it's close to being ready (we can increase the buffer size for the above issue that's fine).